### PR TITLE
Add error checking on sender and recipient arguments.

### PR DESCRIPTION
**Pull request by hurricane766**
I had the situation arise where recipients in sendRaw was undefined and my app would crash in mailgun.js. So I added a couple of checks for sender and recipients. The checks for undefined aren't necessary as they would be caught by the subsequent check for string or array, but I thought it may help to be more explicit in those cases.

### CODE REVIEW

File: mailgun.js [hunk 1]
```diff
@@ -85,6 +85,18 @@ Mailgun.prototype.sendText = function(sender, recipients, subject, text) {
...
+  if (typeof sender === 'undefined')
+    throw new Error('sender is undefined');
+
+  if (typeof sender !== 'string')
+    throw new Error('sender is not a string');
+
+  if (typeof recipients === 'undefined')
+    throw new Error('recipients is undefined');
+
+  if (typeof recipients !== 'string' && typeof recipients !== 'array')
```
Comment (shz) [inline mailgun.js]: `typeof recipients !== 'array'` isn't going to work, as `typeof` an array is `'object'`.  One alternative would be `&& !(recipients instanceof Array)`.  Another alternative would just be to check for `recipients.length !== undefined`.

Either of those, or some other solution will do.  Whichever you prefer :)

File: mailgun.js [hunk 2]
```diff
@@ -140,6 +152,18 @@ Mailgun.prototype.sendRaw = function(sender, recipients, rawBody) {
...
```
Comment (shz) [inline mailgun.js]: Ditto here; `typeof recipients !== 'array'` isn't going to work.

**Comment by shz** [comment]
Thanks for the PR!  No issues aside from the array stuff, I'll merge once that's taken care of.

**Comment by hurricane766** [comment]
I didn't realize that about `typeof`, I guess that explains the need for the `util.isArray` function. Thanks :)

**Comment by shz** [comment]
Yeah, that little nuance has bitten me in the ass more times than I care to admit.  It just seems like it should be there, but nope!

Merging, thanks again.

**Pull request merged**